From 4d8fc9f46e4bedac16bf1ecbb010d6456d4130d0 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Mon, 3 Nov 2014 07:52:25 -0500
Subject: [infiniband] iser: Add TIMEWAIT_EXIT event handling

Message-id: <f31b9c30f2fb2a89fc5a73cb2a7d0d86fa9617fa.1415000301.git.avadai@redhat.com>
Patchwork-id: 98696
O-Subject: [Patch RHEL7.1 27/37] IB/iser: Add TIMEWAIT_EXIT event handling
Bugzilla: 1107622
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Doug Ledford <dledford@redhat.com>

From: Roi Dayan <roid@mellanox.com>

In case the DISCONNECTED event is not delivered after rdma_disconnect
is called, the CM waits TIMEWAIT seconds and delivers the
TIMEWAIT_EXIT local event. We use this as the notification needed to
continue in the teardown and release sequence.

Signed-off-by: Ariel Nahum <arieln@mellanox.com>
Signed-off-by: Roi Dayan <roid@mellanox.com>
Reviewed-by: Sagi Grimberg <sagig@mellanox.com>
Signed-off-by: Or Gerlitz <ogerlitz@mellanox.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 9579d603502d0f24272c4dd70451d97c8d306b54)
Signed-off-by: Amir Vadai <amirv@mellanox.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/ulp/iser/iser_verbs.c b/drivers/infiniband/ulp/iser/iser_verbs.c
index dc0c90f..da6f3dd 100644
--- a/drivers/infiniband/ulp/iser/iser_verbs.c
+++ b/drivers/infiniband/ulp/iser/iser_verbs.c
@@ -785,6 +785,7 @@ static int iser_cma_handler(struct rdma_cm_id *cma_id, struct rdma_cm_event *eve
  case RDMA_CM_EVENT_DISCONNECTED:
  case RDMA_CM_EVENT_DEVICE_REMOVAL:
  case RDMA_CM_EVENT_ADDR_CHANGE:
+ case RDMA_CM_EVENT_TIMEWAIT_EXIT:
   iser_disconnected_handler(cma_id);
   break;
  default:
-- 
1.7.1